plugin/trace/trace.go

```diff
@@ -120,6 +121,10 @@ func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 	span.SetTag(tagType, req.Type())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
 
+	for key, value := range t.tags {
+		span.SetTag(key, value)
```

Comment:
This is a good change. One thing - rather than treating `value` as simple static text, it should be able to be a metadata value. See https://coredns.io/plugins/metadata/

Reply:
@johnbelamaric Thank you for the suggestion!

I've implemented your request based on the `rewrite` example. This functionality is significantly more complex than what I need, and introduces more cross-plugin compile-time dependencies but I could see it would be useful to others.

Let me know if this is what you had in mind, thanks.